### PR TITLE
UniquePointer Implementation

### PR DESCRIPTION
**Pull request by kontrybutor**

### CODE REVIEW

File: homework/unique_ptr/uniquePtr.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    T* m_data;
+
+public:
+    explicit UniquePtr(T* data = nullptr)
+        : m_data(data)
+    {
+    }
+    ~UniquePtr()
+    {
+        if (m_data) {
```
Comment (ziobron) [inline homework/unique_ptr/uniquePtr.hpp:17]: if not needed. delete on nullptr is safe.

File: homework/unique_ptr/uniquePtr.hpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    T& operator*() const { return m_data; };
+    T* operator->() const { return *m_data; };
+
+    T* get() const { return m_data; };
+    T* release() noexcept
+    {
+        T* result = nullptr;
+        std::swap(result, nullptr);
+        return result;
```
Comment (ziobron) [inline homework/unique_ptr/uniquePtr.hpp:48]:
```suggestion
        T* result = nullptr;
        std::swap(result, m_data);
        return result;
```

File: homework/unique_ptr/uniquePtr.hpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    void reset(T* data) noexcept
+    {
+        delete m_data;
+        m_data = nullptr;
+        m_data = data;
+    }
```
Comment (ziobron) [inline homework/unique_ptr/uniquePtr.hpp:55]:
```suggestion
    void reset(T* data = nullptr) noexcept
    {
        delete m_data;
        m_data = data;
    }
```

**Review by ziobron — CHANGES_REQUESTED** [review]
Missing unit tests and CI with valgrind / asan.

File: homework/unique_ptr/uniquePtr.hpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        if (&rhs == this) {
+            return *this;
+        }
+        delete m_data;
+        m_data = rhs.m_data;
+        rhs.m_data = nullptr;
+        return *this;
+    }
+
+    T& operator*() const { return m_data; };
```
Comment (MaciejSalamonski) [inline homework/unique_ptr/uniquePtr.hpp:40]: I suggest you to throw an exception there

**Review by MaciejSalamonski — APPROVED** [review]

File: homework/unique_ptr/uniquePtr.hpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    UniquePtr& operator=(UniquePtr&& rhs)
+    {
+        if (&rhs == this) {
+            return *this;
+        }
+        delete m_data;
+        m_data = rhs.m_data;
+        rhs.m_data = nullptr;
+        return *this;
```
Comment (WBigaj) [inline homework/unique_ptr/uniquePtr.hpp:37]: You are returning *this anyway. Can't you refactor it to not repeat yourself.

**Review by WBigaj — COMMENTED** [review]
I have checked your code and have one question.

File: .github/workflows/manual.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    steps:
+    - uses: actions/checkout@master
+
+    - name: Create build folder
+      working-directory: homework/unique_ptr
+      run: mkdir build
+    - name: Compile
+      working-directory: homework/unique_ptr/build
+      run: |
+        cmake ..
```
Comment (ziobron) [inline .github/workflows/manual.yml:18]: missing CMakeLists.txt to run cmake

**Review by ziobron — CHANGES_REQUESTED** [review]
Missing tests and build system

**Pull request closed without merging**